### PR TITLE
Update ch-1 new version V3

### PR DESCRIPTION
**Pull request by Nadai2010**
# Update ch-1 new version V3

This version is updated with the latest features from v3 and some improvements integrated into the OD Hack up to PR-316 of Scaffold Stark

## Types of change

- [x] Feature
- [x] Enhancement

## Comments (optional)

Cairo contracts are not optimized

### CODE REVIEW

File: .github/workflows/main.yml [hunk 1]
```diff
@@ -49,17 +49,6 @@ jobs:
         run: yarn install --immutable
         working-directory: ./packages/nextjs
 
-      - name: Check Code Format
```
Comment (gianalarcon) [inline .github/workflows/main.yml:52]: We will need this actions. I think we dont need to delete them

File: packages/nextjs/hooks/scaffold-stark/__tests__/useAutoConnect.test.ts [hunk 2]
```diff
@@ -0,0 +1,104 @@
+import { renderHook } from "@testing-library/react";
```
Comment (gianalarcon) [inline packages/nextjs/hooks/scaffold-stark/__tests__/useAutoConnect.test.ts:1]: We don't need test for hooks at this stage. Better we remove tests

**Review by gianalarcon — COMMENTED** [review]
I think this [file](https://github.com/Scaffold-Stark/speedrunstark/pull/152/files#diff-f4a974c1754240e0bc767b9c67b640e3299a32ba251db5de10d9f5383ece311b) is already here on a different folder, no need it again

**Comment by gianalarcon** [comment]
image
The THRESHOLD is not loading on devnet, I expect to see 1ETH as defined on the [SC](https://github.com/Scaffold-Stark/speedrunstark/blob/1867f9d78022764d7b9d7bf453e9ce16f0e0b11a/packages/snfoundry/contracts/src/Staker.cairo#L30) there, as the stable version of this challenge

**Comment by gianalarcon** [comment]
yarn test not working
1. No need to upgrade or improve contracts for now. Just fix the test to match with new version of snforge
Tip: To keep using `assert_eq!` on tests, we have to import new [dependency](https://github.com/The-Marquis-Gaming/themarquis-ui/blob/ec08223f617a93a902e46ca14acb7f8b2d9fd5e2/packages/snfoundry/contracts/Scarb.toml#L17C1-L17C14)

File: packages/snfoundry/contracts/src/Staker.cairo [hunk 3]
```diff
@@ -45,7 +46,7 @@ pub mod Staker {
     #[storage]
     struct Storage {
         eth_token_dispatcher: IERC20CamelDispatcher,
-        balances: LegacyMap<ContractAddress, u256>,
+        balances: Map<ContractAddress, u256>,
```
Comment (ngjupeng) [inline packages/snfoundry/contracts/src/Staker.cairo:49]: the [readme ](https://github.com/Scaffold-Stark/speedrunstark/tree/nadai/challenge-1-update?tab=readme-ov-file#checkpoint-1--staking-) need to update of using Map as well

**Comment by ngjupeng** [comment]
@Nadai2010 can update this [bug](https://github.com/Scaffold-Stark/scaffold-stark-2/pull/323)

**Comment by Nadai2010** [comment]
Yes, i update today

File: packages/nextjs/components/stake/StakeContractInteraction.tsx [hunk 4]
```diff
@@ -49,23 +50,26 @@ export const StakeContractInteraction = ({ address }: { address?: string }) => {
...
+  const { sendAsync: execute } = useScaffoldWriteContract({
     contractName: "Staker",
     functionName: "execute",
   });
-  const { writeAsync: withdrawETH } = useScaffoldWriteContract({
+
+  const { sendAsync: withdrawETH } = useScaffoldWriteContract({
     contractName: "Staker",
     functionName: "withdraw",
+    args: []
```
Comment (Nadai2010) [inline packages/nextjs/components/stake/StakeContractInteraction.tsx:69]: The new changes in useScaffoldWriteContract give me error if I don't add the `args: []`

Captura desde 2024-10-15 22-40-28
Captura desde 2024-10-15 22-39-49

**Comment by Nadai2010** [comment]
Hi, @metalboyrick, @jrcarlos2000, @ngjupeng  I have added the changes to the hook in the [PR329](https://github.com/Scaffold-Stark/scaffold-stark-2/pull/329) and [PR325](https://github.com/Scaffold-Stark/scaffold-stark-2/pull/325).

The challenger-1 is working, only need to resolve the checkSummer of the events.

**Comment by metalboyrick** [comment]
Fixed with Scaffold PR 332: https://github.com/Scaffold-Stark/scaffold-stark-2/pull/332

**Pull request merged**